File: Portfolio/Portfolio.Blazor/Data/Entities/SnippetTranslation.cs
namespace Portfolio.Blazor.Data.Entities;

public sealed class SnippetTranslation
{
    public int Id { get; set; }
    public int SnippetId { get; set; }
    public string Locale { get; set; } = string.Empty;
    public string Title { get; set; } = string.Empty;
    public string? Description { get; set; }
    public DateTime? CreatedAt { get; set; }
    public DateTime? UpdatedAt { get; set; }

    public Snippet? Snippet { get; set; }
}

File: laravel/app/Models/SnippetTranslation.php
<?php

namespace App\Models;

use App\Models\Concerns\Auditable;
use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

class SnippetTranslation extends Model
{
    use Auditable, HasFactory;

    protected $fillable = ['snippet_id', 'locale', 'title', 'description'];

    /**
     * @return BelongsTo<Snippet, $this>
     */
    public function snippet(): BelongsTo
    {
        return $this->belongsTo(Snippet::class);
    }
}
